pkg/kubelet/cm/dra/manager.go

```diff
@@ -123,15 +123,16 @@ func (m *ManagerImpl) PrepareResources(pod *v1.Pod) error {
 		// whose resources are being prepared would never have
 		// started, so it's OK (actually correct) to not include it
 		// in the cache.
-		claimInfo.addPodReference(pod.UID)
+		claimInfoCopy.addPodReference(pod.UID)
 
-		if claimInfo.prepared {
+		if claimInfoCopy.prepared {
 			// Already prepared this claim, no need to prepare it again
+			m.cache.add(&claimInfoCopy)
```

Comment:
It feels weird to call a function named `add` here. Is the purpose of this to overwrite the existing claimInfo entry with the one that has the updated pod reference? Maybe an `update` call would make that clearer?

Reply:
I agree, but this call adds or replaces value in the map. How about addOrUpdate or something similar?

Reply:
`addOrUpdate()` seems reasonable